src/App.js

```diff
@@ -107,56 +115,73 @@ function App() {
         setLoggedIn(true, "mturk", turkId)
         /* eslint-enable */
       }
+      // If prolific
+      else if (PROLIFIC) {
+        const pID = getProlificId();
+        if (FIREBASE && pID) {
+          setMethod("firebase");
+          setLoggedIn(true, "prolific", pID);
+        } else {
+          setReject(true);
+        }
+      }
       // If firebase
       else if (FIREBASE) {
-        // If prolific - currectly placeholder
-        if (PROLIFIC) {
-        }
-        // Otherwise - currently placeholder
-        else {
-        }
         setMethod("firebase");
+      } else {
+        setReject(true);
       }
     }
-  }, [setLoggedIn, startDate, timestamp]);
+  }, [setLoggedIn, startDate]);
```

Comment:
startDate not state

Reply:
looking at the `useCallback` docs, `startDate` is needed there, but it shouldn't be needed here as it will never change

Reply:
I ended up just removing the `useCallback` and made it a normal function. What do you think @suivo2001

Reply:
Yeah if that works then better. Let me know when you push to beads so I can integrate it to msit.

Reply:
Never mind... I think we have to keep it as is, because as a regular function it calls useEffect in a loop, and you can't have an empty dependency array for `useCallback`

Reply:
I think we need `startDate` in the dependency array for `useCallback`, but not for `useEffect` since it's a constant.  Does that work?